frontends/common/options.cpp

```diff
@@ -372,6 +374,20 @@ void CompilerOptions::dumpPass(const char* manager, unsigned seq, const char* pa
     }
 }
 
+std::tuple<cstring, cstring, cstring> CompilerOptions::parseTarget() {
+    std::vector<std::string> splits;
+    std::string target_str(target.c_str());
+    boost::split(splits, target_str, [](char c){return c == '-';});
+    if (splits.size() != 3)
+        BUG("Invalid target %s", target);
```

Comment:
We have never specified as a requirement that target has to look this way. For example, the ebpf back-end has different requirements for the target string.

Reply:
The 3-component target is a feature/requirement of the p4c driver program, so if you want a backend to be callable via the driver, it needs to use the same kind of target.  If that is not adequate for some target, we need to figure out what needs to be added to the p4c driver

Reply:
I have never seen a specification of what the driver is supposed to do.
If the driver specification influences the design of the compiler we have to have it written down somewhere.